File: Android_Studio/cpp_test/first_cpp/app/src/main/java/com/example/first_cpp/MainActivity.java
package com.example.first_cpp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.example.first_cpp.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'first_cpp' library on application startup.
    static {
        System.loadLibrary("first_cpp");
    }

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Example of a call to a native method
        TextView tv = binding.sampleText;
        tv.setText(stringFromJNI());
    }

    /**
     * A native method that is implemented by the 'first_cpp' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();
}

File: Android_Studio/Dice Roller/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState) // 앱이 열릴때, 먼저 호출
        setContentView(R.layout.activity_main)
    }
}